tests/unit/test_charm.py

```diff
@@ -14,25 +14,33 @@ def setUp(self):
         self.harness = Harness(IntegratorCharm)
         self.addCleanup(self.harness.cleanup)
         self.harness.begin()
+        self.peer_relation_id = self.harness.add_relation(
+            "data-integrator-peers", "data-integrator-peers"
+        )
+        self.charm = self.harness.charm
 
     def test_on_start(self):
-        self.harness.charm.on.start.emit()
+        self.harness.set_leader(True)
+        self.charm.on.config_changed.emit()
+        self.charm.on.start.emit()
         # Ensure we set an ActiveStatus with no message
         self.assertEqual(
-            self.harness.model.unit.status, BlockedStatus("The database name is not specified.")
+            self.harness.model.unit.status,
+            BlockedStatus("Please specify either topic or database name"),
         )
 
     def test_action_failures(self):
-        self.harness.update_config({"database": ""})
+        self.harness.set_leader(True)
+        self.harness.update_config({"database-name": None})
         action_event = Mock()
         self.harness.charm._on_get_credentials_action(action_event)
 
         self.assertEqual(
             action_event.fail.call_args,
-            [("The database name is not specified in the config.",)],
+            [("The database name or topic name is not specified in the config.",)],
         )
```

Comment:
**issue:** I would not test based on non-`constants`/`literals` strings, as these could change any time, slowing down refactoring for failing tests. Maybe here check for the *existence* of `action_event.fail` and `action_event.fail.call_args`, making sure there is *some* message, rather than exactly that one.

Reply:
If possible, I'd go with Marc suggestion. Otherwise we can just keep it as is for the time being.